CIScripts/Common/Invoke-NativeCommand.ps1

```diff
@@ -18,30 +21,84 @@ function Invoke-NativeCommand {
     # because Jenkins misinterprets $LastExitCode variable.
     #
     # Note: The command has to return 0 exitcode to be considered successful.
+    #
+    # The wrapper returns a dictionary with a two optional fields:
+    # If -AllowNonZero is set, the .ExitCode contains an exitcode of a command.
+    # If -CaputerOutput is set, the .Output contains captured output
+    # (otherwise, it will be printed usint Write-Host)
+
+    function Invoke-CommandLocalOrRemote {
+        Param(
+            [parameter(Mandatory=$true)] [AllowNull()] [PSSessionT] $Session,
+            [parameter(Mandatory=$true)] [ScriptBlock] $ScriptBlock
+        )
+
+        if ($Session) {
+            Invoke-Command -Session $Session -ScriptBlock $ScriptBlock
+        } else {
+            Invoke-Command -ScriptBlock $ScriptBlock
+        }
+    }
 
-    # If an executable in $ScriptBlock wouldn't be found then while checking $LastExitCode
-    # we would be checking the exit code of a previous command. To avoid this we clear $LastExitCode.
-    $Global:LastExitCode = $null
+    Invoke-CommandLocalOrRemote -Session $Session -ScriptBlock {
+        # If an executable in $ScriptBlock wouldn't be found then while checking $LastExitCode
+        # we would be checking the exit code of a previous command. To avoid this we clear $LastExitCode.
+        $Global:LastExitCode = $null
```

Comment:
Disagree, this would be a low level helper and shouldn't be called after high-level concepts, like repairing. `Clear-ExitCode` would be a better fit here.